test/unit/specs/vislib/visualizations/tile_map.js

```diff
@@ -196,6 +199,229 @@ define(function (require) {
+          });
+        });
+
+        it('should return false if feature is not close enough to event latlng to display tooltip', function () {
+          vis.handler.charts.forEach(function (chart) {
+            var mapData = chart.chartData.geoJson;
+            var i = _.random(0, mapData.features.length - 1);
+            var point = L.latLng(90, -180);
+            var zoom = _.random(1, 12);
+            var feature = mapData.features[i];
+            var map = chart.maps[0];
+            expect(chart.tooltipProximity(point, zoom, feature, map)).to.be(false);
+          });
+        });
+      });
+
+      describe('nearestFeature method', function () {
+        it('should return an object', function () {
+          vis.handler.charts.forEach(function (chart) {
+            var mapData = chart.chartData.geoJson;
+            var i = _.random(0, mapData.features.length - 1);
+            var feature = mapData.features[i];
+            var point = feature.properties.latLng;
+            expect(_.isObject(chart.nearestFeature(point, mapData))).to.be(true);
+          });
+        });
+
+        it('should return a geoJson feature', function () {
+          vis.handler.charts.forEach(function (chart) {
+            var mapData = chart.chartData.geoJson;
+            var i = _.random(0, mapData.features.length - 1);
+            var feature = mapData.features[i];
+            var point = feature.properties.latLng;
+            expect(chart.nearestFeature(point, mapData).type).to.be('Feature');
+          });
+        });
+
+        it('should return the geoJson feature with same latlng as point', function () {
+          vis.handler.charts.forEach(function (chart) {
+            var mapData = chart.chartData.geoJson;
+            var i = _.random(0, mapData.features.length - 1);
+            var feature = mapData.features[i];
+            var point = feature.properties.latLng;
+            expect(chart.nearestFeature(point, mapData)).to.be(feature);
+          });
+        });
+      });
+
+      describe('addLatLng method', function () {
+        it('should add object to properties of each feature', function () {
+          vis.handler.charts.forEach(function (chart) {
+            var map = chart.maps[0];
+            var mapData = chart.addLatLng(chart.chartData.geoJson);
+            var i = _.random(0, mapData.features.length - 1);
+            var feature = mapData.features[i];
+            expect(_.isObject(feature.properties.latLng)).to.be(true);
+          });
+        });
+
+        it('should add latLng with lat to properties of each feature', function () {
+          vis.handler.charts.forEach(function (chart) {
+            var map = chart.maps[0];
+            var mapData = chart.addLatLng(chart.chartData.geoJson);
+            var i = _.random(0, mapData.features.length - 1);
+            var feature = mapData.features[i];
+            expect(_.isFinite(feature.properties.latLng.lat)).to.be(true);
```

Comment:
This test should check the actual value of .lat, not just if its a finite number.